Documentation Fix

DarioMarzella opened the pull request:
Fixed multiple typos in the documentation.
Fixed and updated sample scripts in documentation and tutorial.

The "Create your own features" and "Create your own targets" parts of the advanced tutorial still have to be completely tested and updated.

NicoRenaud commented on docs/deeprank.tools.rst:
```diff
@@ -5,7 +5,6 @@ Tools
 This module contains a series of core tools used for the feature calculations in DeepRank. These tools are at the moment:
 
-    - pdb2sql : a sql based parser of pdb
     - StructureSimilarity : compute irmsd,lrmsd,Fnat and DockQ score (Profit replacement)
```
Structure similarity is also part of pdb2sql now so we should remove that part

NicoRenaud replied:
I actually don't quite know what will be left in that section then

DarioMarzella replied:
.swp and .swo files removed in commit [a504c3c2b7486835d98a681ad710c8372f77c43c](https://github.com/DeepRank/deeprank/commit/a504c3c2b7486835d98a681ad710c8372f77c43c) . 

Do you think we should remove both the docs/deeprank.tools.rst file and the deeprank/tools/README.md since it is [practically empty after I removed pdb2sql from it](https://github.com/DeepRank/deeprank/commit/60b23c2630c344425cc2938f272f8fad19d17c59)?

NicoRenaud replied:
As far as I can say the sasa in tools is only used in NaivePSSM that we shouldn't use. The sparse tool is used a bit so we have to keep it. So for now it's probably best to leave it as is and merge and  close this PR. However we should clean the code a bit and remove the unused parts (like sasa)

DarioMarzella replied:
As far as regards the sasa in particular, I would leave it because I need the sasa calculation (trough freesasa) for implementing the desolvation energy as a feature (Issue #68 ).

NicoRenaud reviewed (changes requested):
Thanks a lot for all the edits ! There are a few .swp files that come from opening a file with VI and not closing it properly. We should not have these files in the repo of course :) So could you remove them before we merge ?

CunliangGeng commented:
Many thanks for the edits!
I tried to generate the doc webpages locally and it gave the following log, and you can see the warnings in the 2nd part of the log, could you fix these warnings?
For the 'No module named" warning (due to the deletion of pdb2sql), you just need to remove the corresponding text from the doc. Thanks!

```
(base) ➜  docs git:(doc_DM) ✗ make html
Running Sphinx v3.1.1
making output directory... done
WARNING: html_static_path entry '_static' does not exist
loading intersphinx inventory from https://docs.python.org/objects.inv...
loading intersphinx inventory from http://docs.scipy.org/doc/numpy/objects.inv...
loading intersphinx inventory from http://pytorch.org/docs/0.3.1/objects.inv...
intersphinx inventory has moved: https://docs.python.org/objects.inv -> https://docs.python.org/3/objects.inv
intersphinx inventory has moved: http://pytorch.org/docs/0.3.1/objects.inv -> https://pytorch.org/docs/0.3.1/objects.inv
intersphinx inventory has moved: http://docs.scipy.org/doc/numpy/objects.inv -> https://numpy.org/doc/stable/objects.inv
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 10 source files that are out of date
updating environment: [new config] 10 added, 0 changed, 0 removed
reading sources... [100%] tutorial_deeplearning


/Users/clgeng/Dropbox/github/deeprank/deeprank/features/AtomicFeature.py:docstring of deeprank.features.AtomicFeature.AtomicFeature:8: WARNING: Unexpected indentation.
/Users/clgeng/Dropbox/github/deeprank/deeprank/features/AtomicFeature.py:docstring of deeprank.features.AtomicFeature.AtomicFeature:14: WARNING: Unexpected indentation.
/Users/clgeng/Dropbox/github/deeprank/deeprank/features/FeatureClass.py:docstring of deeprank.features.FeatureClass.FeatureClass:11: WARNING: Unexpected indentation.
/Users/clgeng/Dropbox/github/deeprank/deeprank/features/FeatureClass.py:docstring of deeprank.features.FeatureClass.FeatureClass:16: WARNING: Unexpected indentation.
/Users/clgeng/Dropbox/github/deeprank/deeprank/generate/NormalizeData.py:docstring of deeprank.generate.NormalizeData.NormParam:11: WARNING: Explicit markup ends without a blank line; unexpected unindent.
WARNING: autodoc: failed to import module 'learn' from module 'deeprank'; the following exception was raised:
No module named 'matplotlib.ticker'
/Users/clgeng/Dropbox/github/deeprank/deeprank/learn/DataSet.py:docstring of deeprank.learn.DataSet.DataSet:33: WARNING: Unexpected indentation.
/Users/clgeng/Dropbox/github/deeprank/deeprank/learn/DataSet.py:docstring of deeprank.learn.DataSet.DataSet:35: WARNING: Unexpected indentation.
/Users/clgeng/Dropbox/github/deeprank/deeprank/learn/DataSet.py:docstring of deeprank.learn.DataSet.DataSet:36: WARNING: Block quote ends without a blank line; unexpected unindent.
/Users/clgeng/Dropbox/github/deeprank/deeprank/learn/DataSet.py:docstring of deeprank.learn.DataSet.DataSet:39: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/clgeng/Dropbox/github/deeprank/deeprank/learn/DataSet.py:docstring of deeprank.learn.DataSet.DataSet:40: WARNING: Definition list ends without a blank line; unexpected unindent.
/Users/clgeng/Dropbox/github/deeprank/deeprank/learn/DataSet.py:docstring of deeprank.learn.DataSet.DataSet:59: WARNING: Unexpected indentation.
/Users/clgeng/Dropbox/github/deeprank/deeprank/learn/DataSet.py:docstring of deeprank.learn.DataSet.DataSet:60: WARNING: Block quote ends without a blank line; unexpected unindent.
/Users/clgeng/Dropbox/github/deeprank/deeprank/learn/DataSet.py:docstring of deeprank.learn.DataSet.DataSet.get_raw_feature_name:9: WARNING: Unexpected indentation.
/Users/clgeng/Dropbox/github/deeprank/deeprank/learn/DataSet.py:docstring of deeprank.learn.DataSet.DataSet.get_input_shape:7: WARNING: Unexpected indentation.
WARNING: autodoc: failed to import module 'learn.NeuralNet' from module 'deeprank'; the following exception was raised:
No module named 'matplotlib.ticker'
WARNING: autodoc: failed to import module 'StructureSimilarity' from module 'deeprank.tools'; the following exception was raised:
No module named 'deeprank.tools.StructureSimilarity'


looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] tutorial_deeplearning
generating indices...  genindex py-modindexdone
highlighting module code... [100%] deeprank.tools.sparse
writing additional pages...  searchdone
copying static files... ... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 18 warnings.

The HTML pages are in _build/html.
```

DarioMarzella commented:
@CunliangGeng I am actually getting completely different logs when I try to generate the documentation pages, I am not sure whether it depends on the older Sphinx version I have on alembick or on something else. 
Even running in python the scripts that raise errors in your log (like AtomicFeatures.py) I get different errors, like the following one (which  just noticed is present in every branch, so I am going to fix it) :
`Traceback (most recent call last):
  File "AtomicFeature.py", line 968, in <module>
    atfeat.sqldb.close()
AttributeError: 'pdb2sql' object has no attribute 'close'`

Here is my Sphinx log:

`Running Sphinx v2.2.0
making output directory... done
WARNING: html_static_path entry '_static' does not exist
loading intersphinx inventory from https://docs.python.org/objects.inv...
intersphinx inventory has moved: https://docs.python.org/objects.inv -> https://docs.python.org/3/objects.inv
loading intersphinx inventory from http://docs.scipy.org/doc/numpy/objects.inv...
intersphinx inventory has moved: http://docs.scipy.org/doc/numpy/objects.inv -> https://numpy.org/doc/stable/objects.inv
loading intersphinx inventory from http://pytorch.org/docs/0.3.1/objects.inv...
intersphinx inventory has moved: http://pytorch.org/docs/0.3.1/objects.inv -> https://pytorch.org/docs/0.3.1/objects.inv
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 10 source files that are out of date
updating environment: [new config] 10 added, 0 changed, 0 removed
reading sources... [100%] tutorial_deeplearning
WARNING: autodoc: failed to import module 'generate' from module 'deeprank'; the following exception was raised:
Traceback (most recent call last):
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/sphinx/ext/autodoc/importer.py", line 32, in import_module
    return importlib.import_module(modname)
  File "/home/dariomarzella/anaconda3/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/dariomarzella/deeprank/deeprank/generate/__init__.py", line 1, in <module>
    from .DataGenerator import DataGenerator
  File "/home/dariomarzella/deeprank/deeprank/generate/DataGenerator.py", line 14, in <module>
    from deeprank.generate import GridTools as gt
  File "/home/dariomarzella/deeprank/deeprank/generate/GridTools.py", line 8, in <module>
    import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/__init__.py", line 54, in <module>
    from .pdb2sqlcore import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/pdb2sqlcore.py", line 7, in <module>
    import pandas as pd
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/__init__.py", line 22, in <module>
    from pandas.compat.numpy import (
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/compat/numpy/__init__.py", line 10, in <module>
    _nlv = LooseVersion(_np_version)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 304, in __init__
    self.parse(vstring)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 312, in parse
    components = [x for x in self.component_re.split(vstring)
TypeError: expected string or bytes-like object

WARNING: autodoc: failed to import module 'generate.DataGenerator' from module 'deeprank'; the following exception was raised:
Traceback (most recent call last):
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/sphinx/ext/autodoc/importer.py", line 32, in import_module
    return importlib.import_module(modname)
  File "/home/dariomarzella/anaconda3/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/dariomarzella/deeprank/deeprank/generate/__init__.py", line 1, in <module>
    from .DataGenerator import DataGenerator
  File "/home/dariomarzella/deeprank/deeprank/generate/DataGenerator.py", line 14, in <module>
    from deeprank.generate import GridTools as gt
  File "/home/dariomarzella/deeprank/deeprank/generate/GridTools.py", line 8, in <module>
    import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/__init__.py", line 54, in <module>
    from .pdb2sqlcore import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/pdb2sqlcore.py", line 7, in <module>
    import pandas as pd
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/__init__.py", line 22, in <module>
    from pandas.compat.numpy import (
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/compat/numpy/__init__.py", line 10, in <module>
    _nlv = LooseVersion(_np_version)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 304, in __init__
    self.parse(vstring)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 312, in parse
    components = [x for x in self.component_re.split(vstring)
TypeError: expected string or bytes-like object

WARNING: autodoc: failed to import module 'generate.NormalizeData' from module 'deeprank'; the following exception was raised:
Traceback (most recent call last):
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/sphinx/ext/autodoc/importer.py", line 32, in import_module
    return importlib.import_module(modname)
  File "/home/dariomarzella/anaconda3/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/dariomarzella/deeprank/deeprank/generate/__init__.py", line 1, in <module>
    from .DataGenerator import DataGenerator
  File "/home/dariomarzella/deeprank/deeprank/generate/DataGenerator.py", line 14, in <module>
    from deeprank.generate import GridTools as gt
  File "/home/dariomarzella/deeprank/deeprank/generate/GridTools.py", line 8, in <module>
    import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/__init__.py", line 54, in <module>
    from .pdb2sqlcore import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/pdb2sqlcore.py", line 7, in <module>
    import pandas as pd
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/__init__.py", line 22, in <module>
    from pandas.compat.numpy import (
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/compat/numpy/__init__.py", line 10, in <module>
    _nlv = LooseVersion(_np_version)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 304, in __init__
    self.parse(vstring)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 312, in parse
    components = [x for x in self.component_re.split(vstring)
TypeError: expected string or bytes-like object

WARNING: autodoc: failed to import module 'generate.GridTools' from module 'deeprank'; the following exception was raised:
Traceback (most recent call last):
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/sphinx/ext/autodoc/importer.py", line 32, in import_module
    return importlib.import_module(modname)
  File "/home/dariomarzella/anaconda3/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/dariomarzella/deeprank/deeprank/generate/__init__.py", line 1, in <module>
    from .DataGenerator import DataGenerator
  File "/home/dariomarzella/deeprank/deeprank/generate/DataGenerator.py", line 14, in <module>
    from deeprank.generate import GridTools as gt
  File "/home/dariomarzella/deeprank/deeprank/generate/GridTools.py", line 8, in <module>
    import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/__init__.py", line 54, in <module>
    from .pdb2sqlcore import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/pdb2sqlcore.py", line 7, in <module>
    import pandas as pd
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/__init__.py", line 22, in <module>
    from pandas.compat.numpy import (
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/compat/numpy/__init__.py", line 10, in <module>
    _nlv = LooseVersion(_np_version)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 304, in __init__
    self.parse(vstring)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 312, in parse
    components = [x for x in self.component_re.split(vstring)
TypeError: expected string or bytes-like object

WARNING: autodoc: failed to import module 'learn' from module 'deeprank'; the following exception was raised:
Traceback (most recent call last):
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/sphinx/ext/autodoc/importer.py", line 32, in import_module
    return importlib.import_module(modname)
  File "/home/dariomarzella/anaconda3/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/dariomarzella/deeprank/deeprank/learn/__init__.py", line 3, in <module>
    from .DataSet import DataSet
  File "/home/dariomarzella/deeprank/deeprank/learn/DataSet.py", line 12, in <module>
    import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/__init__.py", line 54, in <module>
    from .pdb2sqlcore import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/pdb2sqlcore.py", line 7, in <module>
    import pandas as pd
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/__init__.py", line 22, in <module>
    from pandas.compat.numpy import (
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/compat/numpy/__init__.py", line 10, in <module>
    _nlv = LooseVersion(_np_version)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 304, in __init__
    self.parse(vstring)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 312, in parse
    components = [x for x in self.component_re.split(vstring)
TypeError: expected string or bytes-like object

WARNING: autodoc: failed to import module 'learn.DataSet' from module 'deeprank'; the following exception was raised:
Traceback (most recent call last):
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/sphinx/ext/autodoc/importer.py", line 32, in import_module
    return importlib.import_module(modname)
  File "/home/dariomarzella/anaconda3/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/dariomarzella/deeprank/deeprank/learn/__init__.py", line 3, in <module>
    from .DataSet import DataSet
  File "/home/dariomarzella/deeprank/deeprank/learn/DataSet.py", line 12, in <module>
    import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/__init__.py", line 54, in <module>
    from .pdb2sqlcore import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/pdb2sqlcore.py", line 7, in <module>
    import pandas as pd
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/__init__.py", line 22, in <module>
    from pandas.compat.numpy import (
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/compat/numpy/__init__.py", line 10, in <module>
    _nlv = LooseVersion(_np_version)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 304, in __init__
    self.parse(vstring)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 312, in parse
    components = [x for x in self.component_re.split(vstring)
TypeError: expected string or bytes-like object

WARNING: autodoc: failed to import module 'learn.NeuralNet' from module 'deeprank'; the following exception was raised:
Traceback (most recent call last):
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/sphinx/ext/autodoc/importer.py", line 32, in import_module
    return importlib.import_module(modname)
  File "/home/dariomarzella/anaconda3/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/dariomarzella/deeprank/deeprank/learn/__init__.py", line 3, in <module>
    from .DataSet import DataSet
  File "/home/dariomarzella/deeprank/deeprank/learn/DataSet.py", line 12, in <module>
    import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/__init__.py", line 54, in <module>
    from .pdb2sqlcore import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/pdb2sqlcore.py", line 7, in <module>
    import pandas as pd
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/__init__.py", line 22, in <module>
    from pandas.compat.numpy import (
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/compat/numpy/__init__.py", line 10, in <module>
    _nlv = LooseVersion(_np_version)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 304, in __init__
    self.parse(vstring)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 312, in parse
    components = [x for x in self.component_re.split(vstring)
TypeError: expected string or bytes-like object

WARNING: autodoc: failed to import module 'tools' from module 'deeprank'; the following exception was raised:
Traceback (most recent call last):
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/sphinx/ext/autodoc/importer.py", line 32, in import_module
    return importlib.import_module(modname)
  File "/home/dariomarzella/anaconda3/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/dariomarzella/deeprank/deeprank/tools/__init__.py", line 2, in <module>
    from .sasa import SASA
  File "/home/dariomarzella/deeprank/deeprank/tools/sasa.py", line 2, in <module>
    import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/__init__.py", line 54, in <module>
    from .pdb2sqlcore import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/pdb2sqlcore.py", line 7, in <module>
    import pandas as pd
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/__init__.py", line 22, in <module>
    from pandas.compat.numpy import (
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/compat/numpy/__init__.py", line 10, in <module>
    _nlv = LooseVersion(_np_version)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 304, in __init__
    self.parse(vstring)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 312, in parse
    components = [x for x in self.component_re.split(vstring)
TypeError: expected string or bytes-like object

WARNING: autodoc: failed to import module 'tools.StructureSimilarity' from module 'deeprank'; the following exception was raised:
Traceback (most recent call last):
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/sphinx/ext/autodoc/importer.py", line 32, in import_module
    return importlib.import_module(modname)
  File "/home/dariomarzella/anaconda3/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/dariomarzella/deeprank/deeprank/tools/__init__.py", line 2, in <module>
    from .sasa import SASA
  File "/home/dariomarzella/deeprank/deeprank/tools/sasa.py", line 2, in <module>
    import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/__init__.py", line 54, in <module>
    from .pdb2sqlcore import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/pdb2sqlcore.py", line 7, in <module>
    import pandas as pd
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/__init__.py", line 22, in <module>
    from pandas.compat.numpy import (
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/compat/numpy/__init__.py", line 10, in <module>
    _nlv = LooseVersion(_np_version)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 304, in __init__
    self.parse(vstring)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 312, in parse
    components = [x for x in self.component_re.split(vstring)
TypeError: expected string or bytes-like object

WARNING: autodoc: failed to import module 'tools.sasa' from module 'deeprank'; the following exception was raised:
Traceback (most recent call last):
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/sphinx/ext/autodoc/importer.py", line 32, in import_module
    return importlib.import_module(modname)
  File "/home/dariomarzella/anaconda3/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/dariomarzella/deeprank/deeprank/tools/__init__.py", line 2, in <module>
    from .sasa import SASA
  File "/home/dariomarzella/deeprank/deeprank/tools/sasa.py", line 2, in <module>
    import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/__init__.py", line 54, in <module>
    from .pdb2sqlcore import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/pdb2sqlcore.py", line 7, in <module>
    import pandas as pd
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/__init__.py", line 22, in <module>
    from pandas.compat.numpy import (
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/compat/numpy/__init__.py", line 10, in <module>
    _nlv = LooseVersion(_np_version)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 304, in __init__
    self.parse(vstring)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 312, in parse
    components = [x for x in self.component_re.split(vstring)
TypeError: expected string or bytes-like object

WARNING: autodoc: failed to import module 'tools.sparse' from module 'deeprank'; the following exception was raised:
Traceback (most recent call last):
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/sphinx/ext/autodoc/importer.py", line 32, in import_module
    return importlib.import_module(modname)
  File "/home/dariomarzella/anaconda3/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/dariomarzella/deeprank/deeprank/tools/__init__.py", line 2, in <module>
    from .sasa import SASA
  File "/home/dariomarzella/deeprank/deeprank/tools/sasa.py", line 2, in <module>
    import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/__init__.py", line 54, in <module>
    from .pdb2sqlcore import pdb2sql
  File "/home/dariomarzella/pdb2sql/pdb2sql/pdb2sqlcore.py", line 7, in <module>
    import pandas as pd
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/__init__.py", line 22, in <module>
    from pandas.compat.numpy import (
  File "/home/dariomarzella/anaconda3/lib/python3.7/site-packages/pandas/compat/numpy/__init__.py", line 10, in <module>
    _nlv = LooseVersion(_np_version)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 304, in __init__
    self.parse(vstring)
  File "/home/dariomarzella/anaconda3/lib/python3.7/distutils/version.py", line 312, in parse
    components = [x for x in self.component_re.split(vstring)
TypeError: expected string or bytes-like object

looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] tutorial_deeplearning
generating indices...  genindex py-modindexdone
highlighting module code... [100%] deeprank.learn.modelGenerator
writing additional pages...  searchdone
copying static files... ... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 12 warnings.`

CunliangGeng commented:
@DarioMarzella the`pdb2sql` needs to updated to the latest version. Also update `Sphinx` to see if the "failed to import module" warnings still exist.

NicoRenaud reviewed (changes requested):
you still have swp and swo files in docs/ Could you remove those as well ? Thanks !

DarioMarzella commented:
> you still have swp and swo files in docs/ Could you remove those as well ? Thanks !

Done in 992a265, thanks for pointing it out.



> @DarioMarzella the`pdb2sql` needs to updated to the latest version. Also update `Sphinx` to see if the "failed to import module" warnings still exist.

I updated both of them. Still, I got the same error and warnings (pdb2sql error and Sphinx failed to import warning).

I fixed the error raised by AtomicFeature.py by changing line 968 atfeat.sqldb.close() to atfeat.sqldb._close(), but I am not sure it is the proper way (don't know what should be the difference between those two functions).

Not sure how to proceed with the Sphinx. My actual Sphinx version is v3.2.1.
As you can see from the screenshot (not sure you can replicate it since we have different logs), not all the pages are correctly generated
<img width="874" alt="Screenshot 2020-09-16 at 12 30 59" src="https://user-images.githubusercontent.com/55996385/93326063-8efbed80-f818-11ea-8711-d31283c97581.png">

CunliangGeng commented:
@DarioMarzella I committed some changes. Now the 'Failed to import module' warning should be solved by setting the `conf.py` and the webpages should show the content properly.

Let's ignore the `WARNING: Unexpected indentation.`  warnings at this moment. They are associated with the code. Our code has been updated quite a lot and now we should merge the `docs` to let it index the new code (maybe the warnings will be gone automatically).

NicoRenaud commented:
IF the tests are passing it's all good for me

CunliangGeng reviewed (commented):
@DarioMarzella The `docs/_build/html/` does not need to be uploaded. Could you remove them?

DarioMarzella commented:
> @DarioMarzella The `docs/_build/html/` does not need to be uploaded. Could you remove them?

Just did it in 03c63ca

CunliangGeng commented:
@DarioMarzella The travis CI failed since the `master` code is too old to use the latest `pdb2sql`. I'm closing this PR and could you please create a new PR to merge to `development` branch?
